src/main/java/org/openRealmOfStars/player/combat/Combat.java

```diff
@@ -231,42 +231,31 @@ public CombatShip getMostPowerfulShip(PlayerInfo info) {
    */
   public void destroyShip(CombatShip ship) {
     if (fleet1.isShipInFleet(ship.getShip())) {
-      fleet1.removeShip(ship.getShip());
-      ShipStat stat = animation.getShooter().getPlayer()
-          .getShipStatByName(animation.getShooter().getShip().getName());
-      if (stat != null) {
-        stat.setNumberOfKills(stat.getNumberOfKills()+1);
-      }
-      stat = ship.getPlayer().getShipStatByName(ship.getShip().getName());
-      if (stat != null) {
-        stat.setNumberOfLoses(stat.getNumberOfLoses()+1);
-        stat.setNumberOfInUse(stat.getNumberOfInUse()-1);
-      }
-      int indexToDelete = shipList.indexOf(ship);
-      shipList.remove(ship);
-      if (indexToDelete < shipIndex && shipIndex > 0) {
-        shipIndex--;
-      }
+      destroyShipFromFleet(ship, fleet1);
     } else if (fleet2.isShipInFleet(ship.getShip())) {
-      fleet2.removeShip(ship.getShip());
-      ShipStat stat = animation.getShooter().getPlayer()
-          .getShipStatByName(animation.getShooter().getShip().getName());
-      if (stat != null) {
-        stat.setNumberOfKills(stat.getNumberOfKills()+1);
-      }
-      stat = ship.getPlayer().getShipStatByName(ship.getShip().getName());
-      if (stat != null) {
-        stat.setNumberOfLoses(stat.getNumberOfLoses()+1);
-        stat.setNumberOfInUse(stat.getNumberOfInUse()-1);
-      }      
-      int indexToDelete = shipList.indexOf(ship);
-      shipList.remove(ship);
-      if (indexToDelete < shipIndex && shipIndex > 0) {
-        shipIndex--;
-      }
+      destroyShipFromFleet(ship, fleet2);
     }
   }
-  
+
```

Comment:
Java docs could
/**
* Destroy ship for fleet's list.
* @param ship ship to destroy from the fleet
* @param fleet containing the ship
*/